misc/python/materialize/output_consistency/expression/expression_characteristics.py

```diff
@@ -0,0 +1,37 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+from enum import Enum
+
+
+class ExpressionCharacteristics(Enum):
+    """Characteristics describing an `Expression`"""
+
+    NULL = 0
+    # not (NULL or ZERO)
+    NON_EMPTY = 1
+
+    ZERO = 100
+    ONE = 101
+    TINY_VALUE = 102
```

Comment:
As a future enhancement, also consider values that are:
- just below 0.5
- just above 0.5
- just below 1

For the purpose of sussing out any issues with rounding, ceiling, fooring and casting.

Reply:
✅ https://github.com/MaterializeInc/materialize/pull/19013/commits/5a5400e379e85caeda0da92341074b8b706bd893